collect.py

```diff
@@ -222,10 +224,27 @@ def get_domtree(driver, file_path):
         f.write(driver.execute_script('return document.documentElement.outerHTML'))
 
 
+def get_coordinates(driver, file_path):
+    dom_tree = etree.HTML(driver.execute_script('return document.documentElement.outerHTML'))
+    dom_element_tree = etree.ElementTree(dom_tree)
+    loc_dict = {}
+    for elem in dom_tree.iter(tag=etree.Element):
+        xpath = dom_element_tree.getpath(elem)
+        try:
+            web_elem = driver.find_element_by_xpath(xpath)
```

Comment:
We do we need this?

Reply:
I know all xpaths should be there. But while testing I did find some cases where driver couldn't find the element. So I had to put this, otherwise it would just leave all other elements.

Reply:
How often does it happen? Can you check with a few websites (say 100)?
Maybe we can implement it in another way and not use lxml here.

Reply:
Its not that often! Even if it is, it is generally for `Scalable Vector Graphics` or `noscript` and that is less than 0.1%. More than 50% of these locations will not ignored in xperts implementation. I will provide with experimental stats in a while.

Reply:
@marco-c we can implement it in another way, but we will need a dictionary of the form {xpath : size, location}. Since lxml is used for parsing the DOM in xpert xpaths should be same.

Reply:
If we want to have all elements I feel there is only one possible way, ie for all elements found in selenium we create the xpath by ourselves.

Reply:
I implemented getting xpath from the selenium web element, this is a significant improvement in speed in comparison to present approach. 
1. While Implementing this I made the xpaths same as given by lxml, problem with `Scalable Vector Graphics` is solved.
2.  `no script` element is only present in lxml dom tree and no element exists corresponding to it in selenium elements acquired by `driver.find_elements_by_css_selector('*')`
3. There are some elements which have which have a `div`in dom's xpath while in xpath got from element we get `div'`
4. There are some elements which have `body` in dom's xpath while `head` in xpath got from element.
5. In some cases there are more number of elements found using element xpath than xpath got from dom. (strange)

Presently in 33 test cases 31 matched perfectly.
1 has 4th issue
1 has 5th issue (which is useless because we will be comparing elements we get from lxml dom)

Reply:
[out.txt](https://github.com/marco-c/autowebcompat/files/2216407/out.txt)

Error is actually very less as xpert ignores small children. 
If you feel this error is tolerable I will continue to work on it. Else I will have to find something different.

Reply:
Number of mismatch 13
Number of chrome 42
Number of firefox 60
Total Number 102
From above text file

Reply:
I think the error is tolerable! Let me review the current code